soh/soh/Enhancements/crowd-control/CrowdControl.cpp

```diff
@@ -0,0 +1,524 @@
+                    dataSend["id"] = packet->packetId;
+                    dataSend["type"] = 0;
+                    dataSend["timeRemaining"] = packet->timeRemaining;
+                    dataSend["status"] = EffectResult::Success;
+
+                    std::string jsonResponse = dataSend.dump();
+                    SDLNet_TCP_Send(tcpsock, const_cast<char*> (jsonResponse.data()), jsonResponse.size() + 1);
+                }
+
+                return;
+            }
+
+            // Decrement remaining time by the second that elapsed between thread runs
+            packet->timeRemaining -= 1000;
+            // We don't want to emit repeated events, so ensure we only emit changes in events
+            if (effectResult != lastResult) {
+                lastResult = effectResult;
+
+                nlohmann::json dataSend;
+                dataSend["id"] = packet->packetId;
+                dataSend["type"] = 0;
+                dataSend["timeRemaining"] = packet->timeRemaining;
+                dataSend["status"] = effectResult;
+
+                std::string jsonResponse = dataSend.dump();
+                SDLNet_TCP_Send(tcpsock, const_cast<char*> (jsonResponse.data()), jsonResponse.size() + 1);
+            }
+        } else if (effectResult == EffectResult::Retry && paused == 0 && packet->timeRemaining > 0) {
+            paused = 1;
+            nlohmann::json dataSend;
+            dataSend["id"] = packet->packetId;
+            dataSend["type"] = 0;
+            dataSend["timeRemaining"] = packet->timeRemaining;
+            dataSend["status"] = EffectResult::Paused;
+
+            std::string jsonResponse = dataSend.dump();
+            SDLNet_TCP_Send(tcpsock, const_cast<char*> (jsonResponse.data()), jsonResponse.size() + 1);
+        }
+        
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+}
+
+void CrowdControl::ReceiveFromCrowdControl()
+{
+    printf("Waiting for a connection from Crowd Control...");
+
+    while (!connected && CVar_GetS32("gCrowdControl", 0)) {
+        tcpsock = SDLNet_TCP_Open(&ip);
+
+        if (tcpsock) {
+            connected = true;
+            printf("Connected to Crowd Control!");
+            break;
+        }
+    }
+
+    while (connected && CVar_GetS32("gCrowdControl", 0) && tcpsock) {
+        int len = SDLNet_TCP_Recv(tcpsock, &received, 512);
+
+        if (!len || !tcpsock || len == -1) {
+            printf("SDLNet_TCP_Recv: %s\n", SDLNet_GetError());
+            break;
+        }
+
+        nlohmann::json dataReceived = nlohmann::json::parse(received);
+
+        CCPacket* packet = new CCPacket();
+        packet->packetId = dataReceived["id"];
+        auto parameters = dataReceived["parameters"];
+        if (parameters.size() > 0) {
+            packet->effectValue = dataReceived["parameters"][0];
+        }
+        packet->effectType = dataReceived["code"].get<std::string>();
+
+        if (strcmp(packet->effectType.c_str(), "high_gravity") == 0 ||
```

Comment:
is there a specific reason to `strcmp` here instead of using built in `std::string` compares? i'd think since `effectType` is a `std::string` we would just want to do
```
if (packet->effectType == "high_gravity")
```